bitcoin/src/p2p/address.rs

```diff
@@ -138,6 +138,67 @@ pub enum AddrV2 {
     Unknown(u8, Vec<u8>),
 }
 
+/// Error types for [`AddrV2`] to [`SocketAddr`] conversion.
+#[derive(Debug, PartialEq, Eq)]
+pub enum AddrV2ConversionError {
+    /// A [`AddrV2::TorV3`] address cannot be converted to a [`SocketAddr`].
+    TorV3NotSupported,
+    /// A [`AddrV2::I2p`] address cannot be converted to a [`SocketAddr`].
+    I2pNotSupported,
+    /// A [`AddrV2::Cjdns`] address can be converted to a [`SocketAddr`],
+    /// but it won't work with a tradicional socket API.
+    CjdnsNotRecommended,
+    /// A [`AddrV2::Unknown`] address cannot be converted to a [`SocketAddr`].
+    UnknownNotSupported,
+}
+
+impl fmt::Display for AddrV2ConversionError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::TorV3NotSupported => write!(f, "TorV3 addresses cannot be converted to SocketAddr"),
+            Self::I2pNotSupported => write!(f, "I2P addresses cannot be converted to SocketAddr"),
+            Self::CjdnsNotRecommended => write!(f, "CJDNS addresses can be converted to SocketAddr, but won't work with a traditional socket API"),
+            Self::UnknownNotSupported => write!(f, "Unknown address type cannot be converted to SocketAddr"),
+        }
+    }
+}
+
+impl std::error::Error for AddrV2ConversionError {}
```

Comment:
In 64387f566e05ee88491121fbe9e44e6fcfbc38e4:

This needs to be feature-gated on `std`. (I assume/hope CI will catch this.)

Reply:
Oh, I see, the whole `address` module is p2p-gated. Ok, never mind.